### PR TITLE
Tech debt/soap ut and other small fixes

### PR DESCRIPTION
**Pull request by aljcalandra**

### CODE REVIEW

File: test/GUI/test_gui.py [hunk 1]
```diff
@@ -6,133 +6,114 @@
...
 
-    @patch('SeleniumLibrary.WaitingKeywords.wait_until_page_contains_element')
-    @patch('SeleniumLibrary.FrameKeywords.select_frame')
-    def test_unselect_and_select_frame_simple(self, select_frame, robot_call):
+    def test_unselect_and_select_frame_simple(self):
         mock_gui = Mock()
         GUILibrary.unselect_and_select_frame(mock_gui, "some_locator")
-        assert robot_call.called_with(15, 1, "Set Focus To Element", "some_locator")
-        assert select_frame.called_with("some_locator")
+        mock_gui.wait_for_and_select_frame.assert_called_with("some_locator")
```
Comment (m-hintz-42) [inline test/GUI/test_gui.py]: Looking at this method I think these are backwards?

The method calls `unselect_frame()` and then `wait_for_and_select_frame()`. Your Unit Test calls these in reverse.

File: test/GUI/test_gui.py [hunk 2]
```diff
@@ -6,133 +6,114 @@
...
         assert mock_gui.unselect_frame.called
```
Comment (m-hintz-42) [inline test/GUI/test_gui.py]: Should be changed to `mock_gui.unselect_frame.assert_called` I believe.

File: test/GUI/test_gui.py [hunk 3]
```diff
@@ -6,133 +6,114 @@
...
-    def test_wait_for_mouse_over_and_click_simple(self, robot_call, mouse_over, click_element):
+    def test_wait_for_mouse_over_and_click_simple(self):
         mock_gui = Mock()
         GUILibrary.wait_for_and_mouse_over_and_click(mock_gui, "some_locator")
-        assert robot_call.called_with(15, 1, "Set Focus To Element", "some_locator")
-        assert mouse_over.called_with("some_locator")
-        assert click_element.called_with("some_locator")
+        mock_gui.wait_for_and_focus_on_element.assert_called_with("some_locator")
+        mock_gui.mouse_over.assert_called_with("some_locator")
+        mock_gui.click_element.assert_called_with("some_locator")
```
Comment (m-hintz-42) [inline test/GUI/test_gui.py]: Why would `click_element` be called in this test? 

I don't believe it is called in the `wait_for_and_mouse_over` method.

File: test/GUI/test_gui.py (hunk 1 above)
Comment (aljcalandra) [reply test/GUI/test_gui.py]: after the method has been called each of those methods is stored in methods_called for the respective mock object. so it's basically just doing a lookup after the fact.

File: test/GUI/test_gui.py (hunk 2 above)
Comment (aljcalandra) [reply test/GUI/test_gui.py]: this one is correct, I'll fix that shortly

File: test/GUI/test_gui.py (hunk 3 above)
Comment (aljcalandra) [reply test/GUI/test_gui.py]: this is wait_for_mouse_over_and_click

Comment (m-hintz-42) [reply test/GUI/test_gui.py]: Lol oops.

**Pull request merged**